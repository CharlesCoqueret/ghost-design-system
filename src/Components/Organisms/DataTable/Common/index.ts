export { ColumnType, SortDirectionEnum } from './types';
export type {
  IButtonCellProps,
  IColumnType,
  IExtraEditableDataTableProps,
  IExtraLineEditableInPlaceDataTableProps,
  IExtraStaticDataTableProps,
  TableType,
} from './types';
