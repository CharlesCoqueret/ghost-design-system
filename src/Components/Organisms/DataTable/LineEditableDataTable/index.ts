export type { ILineEditableDataTableProps } from './LineEditableDataTable';
export { default as LineEditableDataTable } from './LineEditableDataTable';
