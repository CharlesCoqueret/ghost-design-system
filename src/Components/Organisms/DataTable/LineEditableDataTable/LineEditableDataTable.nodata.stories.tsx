import React, { ReactElement, useState } from 'react';
import { ComponentStory, ComponentMeta } from '@storybook/react';
import * as yup from 'yup';

import LineEditableDataTable, { ILineEditableDataTableProps } from './LineEditableDataTable';
import { ColumnType, IColumnType } from '../Common/types';
import { Icon } from '../../../Atoms/Icon';

export default {
  title: 'Organism/DataTable/LineEditableDataTable',
  component: LineEditableDataTable,
  parameters: { actions: { argTypesRegex: '^on.*' }, controls: { sort: 'requiredFirst' } },
} as ComponentMeta<typeof LineEditableDataTable>;

interface IDemoType {
  dynamicSearch: string;
  id: string;
  name?: string;
  status?: string;
  price?: number;
  parts?: number;
  startDate?: Date;
}

const initialData = [];

const validationSchema: yup.SchemaOf<IDemoType> = yup.object({
  dynamicSearch: yup.mixed().optional(),
  hidden: yup.string().required(),
  id: yup.string().required(),
  name: yup.string().required(),
  status: yup.string().required(),
  price: yup.number().required(),
  parts: yup.number().required(),
  startDate: yup.date().min(new Date(), 'start date must be after today').required(),
});

const columns: IColumnType<IDemoType>[] = [
  {
    title: 'Code',
    dataIndex: 'id',
    sorter: true,
    type: ColumnType.CODE,
    hiddenInForm: true,
  },
  {
    title: 'Text',
    dataIndex: 'name',
    type: ColumnType.TEXT,
    ellipsis: true,
    sorter: true,
    editable: true,
  },
  {
    title: 'Badge',
    dataIndex: 'status',
    sorter: true,
    type: ColumnType.BADGE,
    editable: true,
    options: [],
  },
  {
    title: 'Amount',
    dataIndex: 'price',
    sorter: true,
    type: ColumnType.AMOUNT,
    currency: 'â‚¬',
    editable: true,
  },
  {
    title: 'Dynamic Search',
    dataIndex: 'dynamicSearch',
    type: ColumnType.DYNAMICSEARCH,
    editable: true,
    resolveValue: () => {
      return Promise.resolve(undefined);
    },
    noOptionsMessage: () => {
      return 'No results';
    },
    searchOptions: () => {
      return Promise.resolve([]);
    },
  },
  {
    title: 'Percentage',
    dataIndex: 'parts',
    sorter: true,
    type: ColumnType.PERCENTAGE,
    editable: true,
  },
  {
    title: 'Date',
    dataIndex: 'startDate',
    sorter: true,
    type: ColumnType.DATE,
    editable: true,
  },
];

const Template: ComponentStory<(args: ILineEditableDataTableProps<IDemoType>) => ReactElement> = (
  args: ILineEditableDataTableProps<IDemoType>,
) => {
  const [loading, setLoading] = useState(true);

  setTimeout(() => {
    setLoading(false);
  }, 5000);

  return (
    <LineEditableDataTable<IDemoType>
      {...args}
      loading={loading ? <Icon icon={['fal', 'spinner']} size='3x' /> : undefined}
    />
  );
};

export const NoDataAfter5Sec = Template.bind({});
NoDataAfter5Sec.args = {
  columns: columns,
  data: initialData,
  extra: {
    validationSchema: validationSchema,
  },
};
