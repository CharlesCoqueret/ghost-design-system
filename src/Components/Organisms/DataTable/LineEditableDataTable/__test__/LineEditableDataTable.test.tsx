import React from 'react';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import * as yup from 'yup';

// Mocking suneditor which is problematic with Jest
jest.mock('suneditor', () => {});
jest.mock('suneditor/src/plugins/', () => {});
jest.mock('suneditor/src/plugins/submenu/align', () => {});
jest.mock('suneditor/src/plugins/command/blockquote', () => {});
jest.mock('suneditor/src/plugins/submenu/fontColor', () => {});
jest.mock('suneditor/src/plugins/submenu/fontSize', () => {});
jest.mock('suneditor/src/plugins/submenu/formatBlock', () => {});
jest.mock('suneditor/src/plugins/submenu/hiliteColor', () => {});
jest.mock('suneditor/src/plugins/submenu/horizontalRule', () => {});
jest.mock('suneditor/src/plugins/dialog/image', () => {});
jest.mock('suneditor/src/plugins/dialog/link', () => {});
jest.mock('suneditor/src/plugins/submenu/lineHeight', () => {});
jest.mock('suneditor/src/plugins/submenu/list', () => {});
jest.mock('suneditor/src/plugins/submenu/paragraphStyle', () => {});
jest.mock('suneditor/src/plugins/submenu/table', () => {});
jest.mock('suneditor-react', () => {});
jest.mock('suneditor-react/dist', () => {});
jest.mock('suneditor-react/dist/types/lang', () => {});

import LineEditableDataTable from '../LineEditableDataTable';
import { ColumnType, SortDirectionEnum } from '../../Common';

describe('LineEditableDataTable component', () => {
  it('LineEditableDataTable renders', () => {
    const { container } = render(
      <LineEditableDataTable<{ number: number }>
        columns={[{ dataIndex: 'number', title: 'Number', type: ColumnType.NUMBER }]}
        data={[{ number: 1 }]}
        extra={{
          validationSchema: yup.object({
            number: yup.number().required(),
          }),
        }}
      />,
    );

    expect(container).toMatchSnapshot();
  });

  it('LineEditableDataTable handles edit and submit', () => {
    const onRowEditMock = jest.fn();
    const onRowSubmitMock = jest.fn();

    const { container } = render(
      <LineEditableDataTable<{ number: number }>
        columns={[
          { dataIndex: 'number', editable: true, placeholder: 'NUMBER', title: 'Number', type: ColumnType.NUMBER },
        ]}
        data={[{ number: 1 }]}
        extra={{
          isEditable: () => true,
          onRowEdit: onRowEditMock,
          onRowSubmit: onRowSubmitMock,
          validationSchema: yup.object({
            number: yup.number().required(),
          }),
        }}
      />,
    );

    expect(container).toMatchSnapshot();

    userEvent.tab();
    userEvent.keyboard('{Enter}');

    expect(onRowEditMock).toBeCalledTimes(1);
    expect(onRowEditMock).toBeCalledWith({ number: 1 }, 0);

    const input = screen.getByPlaceholderText('NUMBER');
    userEvent.clear(input);
    userEvent.type(input, '2');
    userEvent.tab();
    userEvent.tab();
    userEvent.keyboard('{Enter}');

    expect(onRowSubmitMock).toBeCalledTimes(1);
    expect(onRowSubmitMock).toBeCalledWith({ number: 2 }, 0);
  });

  it('LineEditableDataTable handles non editable content', () => {
    const onRowEditMock = jest.fn();
    const onRowSubmitMock = jest.fn();

    const { container } = render(
      <LineEditableDataTable<{ number: number }>
        columns={[
          { dataIndex: 'number', editable: false, placeholder: 'NUMBER', title: 'Number', type: ColumnType.NUMBER },
        ]}
        data={[{ number: 1 }]}
        extra={{
          isEditable: () => false,
          localization: {
            actionColumn: 'actionColumn',
            editButton: 'editButton',
            modalTitle: 'modalTitle',
            moreActionsMessage: 'moreActionsMessage',
          },
          onRowEdit: onRowEditMock,
          onRowSubmit: onRowSubmitMock,
          validationSchema: yup.object({
            number: yup.number().required(),
          }),
        }}
      />,
    );

    expect(container).toMatchSnapshot();
  });

  it('LineEditableDataTable handles delete', () => {
    const onRowDeleteMock = jest.fn();

    const { container } = render(
      <LineEditableDataTable<{ number: number }>
        columns={[
          { dataIndex: 'number', editable: true, placeholder: 'NUMBER', title: 'Number', type: ColumnType.NUMBER },
        ]}
        data={[{ number: 1 }]}
        extra={{
          isDeletable: () => true,
          onRowDelete: onRowDeleteMock,
          validationSchema: yup.object({
            number: yup.number().required(),
          }),
        }}
      />,
    );

    expect(container).toMatchSnapshot();

    userEvent.tab();
    userEvent.keyboard('{Enter}');

    userEvent.tab();
    userEvent.tab();
    userEvent.keyboard('{Enter}');

    expect(onRowDeleteMock).toBeCalledTimes(1);
    expect(onRowDeleteMock).toBeCalledWith({ number: 1 }, 0);
  });
  it('LineEditableDataTable handles delete with previous defined editedrowindex', () => {
    const onRowDeleteMock = jest.fn();

    const { container } = render(
      <LineEditableDataTable<{ number: number }>
        columns={[
          { dataIndex: 'number', editable: true, placeholder: 'NUMBER', title: 'Number', type: ColumnType.NUMBER },
        ]}
        data={[{ number: 1 }, { number: 2 }]}
        extra={{
          editedRowIndex: 1,
          isDeletable: () => true,
          onRowDelete: onRowDeleteMock,
          validationSchema: yup.object({
            number: yup.number().required(),
          }),
        }}
      />,
    );

    expect(container).toMatchSnapshot();
  });

  it('LineEditableDataTable handles non deletable', () => {
    const onRowDeleteMock = jest.fn();

    const { container } = render(
      <LineEditableDataTable<{ number: number }>
        columns={[
          { dataIndex: 'number', editable: true, placeholder: 'NUMBER', title: 'Number', type: ColumnType.NUMBER },
        ]}
        data={[{ number: 1 }]}
        extra={{
          isDeletable: () => false,
          localization: {
            deleteButton: 'deleteButton',
            deletePopoverMessage: 'deletePopoverMessage',
            deletePopoverCancel: 'deletePopoverCancel',
            deletePopoverConfirm: 'deletePopoverConfirm',
          },
          onRowDelete: onRowDeleteMock,
          validationSchema: yup.object({
            number: yup.number().required(),
          }),
        }}
      />,
    );

    expect(container).toMatchSnapshot();
  });

  it('LineEditableDataTable handles download', () => {
    const onRowDownloadMock = jest.fn();

    const { container } = render(
      <LineEditableDataTable<{ number: number }>
        columns={[{ dataIndex: 'number', placeholder: 'NUMBER', title: 'Number', type: ColumnType.NUMBER }]}
        data={[{ number: 1 }]}
        extra={{
          isDownloadable: () => true,
          onRowDownload: onRowDownloadMock,
          validationSchema: yup.object({
            number: yup.number().required(),
          }),
        }}
      />,
    );

    expect(container).toMatchSnapshot();

    userEvent.tab();
    userEvent.keyboard('{Enter}');

    expect(onRowDownloadMock).toBeCalledTimes(1);
    expect(onRowDownloadMock).toBeCalledWith({ number: 1 }, 0);
  });

  it('LineEditableDataTable handles non downloadable', () => {
    const onRowDownloadMock = jest.fn();

    const { container } = render(
      <LineEditableDataTable<{ number: number }>
        columns={[{ dataIndex: 'number', placeholder: 'NUMBER', title: 'Number', type: ColumnType.NUMBER }]}
        data={[{ number: 1 }]}
        extra={{
          isDownloadable: () => false,
          localization: {
            downloadButton: 'downloadButton',
          },
          onRowDownload: onRowDownloadMock,
          validationSchema: yup.object({
            number: yup.number().required(),
          }),
        }}
      />,
    );

    expect(container).toMatchSnapshot();
  });

  it('LineEditableDataTable renders with no data', () => {
    const { container } = render(
      <LineEditableDataTable<{ number: number }>
        columns={[{ dataIndex: 'number', title: 'Number', type: ColumnType.NUMBER }]}
        data={[]}
        extra={{
          validationSchema: yup.object({
            number: yup.number().required(),
          }),
        }}
      />,
    );

    expect(container).toMatchSnapshot();
  });

  it('LineEditableDataTable renders while loading', () => {
    const { container } = render(
      <LineEditableDataTable<{ number: number }>
        columns={[{ dataIndex: 'number', title: 'Number', type: ColumnType.NUMBER }]}
        data={[]}
        extra={{
          validationSchema: yup.object({
            number: yup.number().required(),
          }),
        }}
        loading={<>Loading</>}
      />,
    );

    expect(container).toMatchSnapshot();
  });

  it('LineEditableDataTable handles sort', () => {
    const onSortChangeMock = jest.fn();

    const { container } = render(
      <LineEditableDataTable<{ number: number }>
        columns={[{ dataIndex: 'number', sorter: true, title: 'Number', type: ColumnType.NUMBER }]}
        data={[]}
        extra={{
          validationSchema: yup.object({
            number: yup.number().required(),
          }),
        }}
        loading={<>Loading</>}
        onSortChange={onSortChangeMock}
      />,
    );

    expect(container).toMatchSnapshot();

    userEvent.tab();
    userEvent.keyboard('{Enter}');

    expect(onSortChangeMock).toBeCalledTimes(1);
    expect(onSortChangeMock).toBeCalledWith('number', SortDirectionEnum.DESC);

    userEvent.keyboard('{Enter}');

    expect(onSortChangeMock).toBeCalledTimes(2);
    expect(onSortChangeMock).toBeCalledWith('number', SortDirectionEnum.ASC);

    userEvent.keyboard('{Enter}');

    expect(onSortChangeMock).toBeCalledTimes(3);
    expect(onSortChangeMock).toBeCalledWith();
  });

  it('LineEditableDataTable handles add new line with onNewLine', () => {
    const onNewLineMock = jest.fn().mockImplementation(() => {
      return { number: 0 };
    });
    const onRowCancelEditMock = jest.fn();

    const { container } = render(
      <LineEditableDataTable<{ number: number }>
        columns={[
          { dataIndex: 'number', editable: true, placeholder: 'PLACEHOLDER', title: 'Number', type: ColumnType.NUMBER },
        ]}
        data={[]}
        extra={{
          canAddNewLine: () => true,
          localization: {
            addRow: 'addRow',
            modalTitle: () => 'modalTitle',
          },
          onNewLine: onNewLineMock,
          onRowCancelEdit: onRowCancelEditMock,
          validationSchema: yup.object({
            number: yup.number().required(),
          }),
        }}
      />,
    );

    expect(container).toMatchSnapshot();

    userEvent.tab();
    userEvent.keyboard('{Enter}');

    expect(onNewLineMock).toBeCalledTimes(1);
    expect(onNewLineMock).toBeCalledWith();

    const input = screen.getByPlaceholderText('PLACEHOLDER');
    userEvent.click(input);
    userEvent.tab();
    userEvent.keyboard('{Enter}');

    expect(onRowCancelEditMock).toBeCalledTimes(1);
  });

  it('LineEditableDataTable handles add new line without onNewLine', () => {
    const onRowCancelEditMock = jest.fn();
    console.error = jest.fn();

    const { container } = render(
      <LineEditableDataTable<{ number: number }>
        columns={[
          { dataIndex: 'number', editable: true, placeholder: 'PLACEHOLDER', title: 'Number', type: ColumnType.NUMBER },
        ]}
        data={[]}
        extra={{
          canAddNewLine: () => true,
          onRowCancelEdit: onRowCancelEditMock,
          validationSchema: yup.object({
            number: yup.number().required(),
          }),
        }}
      />,
    );

    expect(container).toMatchSnapshot();

    userEvent.tab();
    userEvent.keyboard('{Enter}');

    expect(console.error).toBeCalledTimes(1);
    expect(console.error).toBeCalledWith('Missing onNewLine function');
  });

  it('LineEditableDataTable with custom modal button', () => {
    const onClickMock = jest.fn().mockImplementation(async () => {
      return Promise.resolve();
    });
    const onNewLineMock = jest.fn().mockImplementation(() => {
      return { number: 0 };
    });

    const { container } = render(
      <LineEditableDataTable<{ number: number }>
        columns={[
          { dataIndex: 'number', editable: true, placeholder: 'PLACEHOLDER', title: 'Number', type: ColumnType.NUMBER },
        ]}
        data={[]}
        extra={{
          canAddNewLine: () => true,
          localization: {
            modalTitle: 'modalTitle',
          },
          onNewLine: onNewLineMock,
          rowEditExtraActions: () => [
            {
              onClick: onClickMock,
              label: 'LABEL',
            },
          ],
          validationSchema: yup.object({
            number: yup.number().required(),
          }),
        }}
      />,
    );

    expect(container).toMatchSnapshot();

    userEvent.tab();
    userEvent.keyboard('{Enter}');

    expect(onNewLineMock).toBeCalledTimes(1);
    expect(onNewLineMock).toBeCalledWith();

    const input = screen.getByPlaceholderText('PLACEHOLDER');
    userEvent.click(input);
    userEvent.tab();
    userEvent.keyboard('{Enter}');

    expect(onClickMock).toBeCalledTimes(1);
  });
});
