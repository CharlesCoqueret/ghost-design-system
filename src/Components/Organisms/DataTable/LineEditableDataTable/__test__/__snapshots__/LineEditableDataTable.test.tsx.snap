// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LineEditableDataTable component LineEditableDataTable handles add new line with onNewLine 1`] = `
<div>
  <div>
    <table
      class="gds-table"
    >
      <thead>
        <tr>
          <th>
            <div
              class="table--header-value"
              tabindex="-1"
            >
              <span
                class="table--header-value--title right"
              >
                Number
              </span>
            </div>
          </th>
        </tr>
      </thead>
      <tbody>
        <tr
          class="no-data"
        >
          <td
            colspan="1"
          >
            <div
              class="no-data-container"
            >
              <div
                class="no-data-text"
              >
                No data
              </div>
              <i
                class="no-data-icon icon"
                icon="fal,inbox"
              />
            </div>
          </td>
        </tr>
      </tbody>
    </table>
    <button
      class="gds-button-content gds-table-new-line"
      color="primary"
      tabindex="0"
      type="button"
    >
      <div
        class="button-label-container"
      >
        addRow
      </div>
    </button>
  </div>
</div>
`;

exports[`LineEditableDataTable component LineEditableDataTable handles add new line without onNewLine 1`] = `
<div>
  <div>
    <table
      class="gds-table"
    >
      <thead>
        <tr>
          <th>
            <div
              class="table--header-value"
              tabindex="-1"
            >
              <span
                class="table--header-value--title right"
              >
                Number
              </span>
            </div>
          </th>
        </tr>
      </thead>
      <tbody>
        <tr
          class="no-data"
        >
          <td
            colspan="1"
          >
            <div
              class="no-data-container"
            >
              <div
                class="no-data-text"
              >
                No data
              </div>
              <i
                class="no-data-icon icon"
                icon="fal,inbox"
              />
            </div>
          </td>
        </tr>
      </tbody>
    </table>
    <button
      class="gds-button-content gds-table-new-line"
      color="primary"
      tabindex="0"
      type="button"
    >
      <div
        class="button-label-container"
      >
        Add row
      </div>
    </button>
  </div>
</div>
`;

exports[`LineEditableDataTable component LineEditableDataTable handles delete 1`] = `
<div>
  <div>
    <table
      class="gds-table"
    >
      <thead>
        <tr>
          <th>
            <div
              class="table--header-value"
              tabindex="-1"
            >
              <span
                class="table--header-value--title right"
              >
                Number
              </span>
            </div>
          </th>
          <th>
            <div
              class="table--header-value"
              tabindex="-1"
            >
              <span
                class="table--header-value--title center"
              >
                Actions
              </span>
            </div>
          </th>
        </tr>
      </thead>
      <tbody>
        <tr
          class=""
          tabindex="-1"
        >
          <td
            class=""
          >
            <div
              class="gds-field-group"
            >
              <span
                class="gds-amount-field amount-field-read-only table--cell-value--number-readonly"
                id="number"
                name="number"
              >
                1
              </span>
            </div>
          </td>
          <td
            class=""
          >
            <div
              class="table--cell--value--button"
            >
              <button
                class="gds-button-content"
                color="reversed"
                tabindex="0"
                type="button"
              >
                <div
                  class="button-icon-container"
                >
                  <i
                    class="button-icon icon"
                    icon="fal,trash-alt"
                  />
                </div>
              </button>
              <div>
                Delete
              </div>
              <div>
                <div>
                  ControlleMenu
                  closed
                  <div
                    class="gds-popover-container"
                  >
                    <div
                      class="popover-title"
                    >
                      Delete?
                    </div>
                    <div
                      class="popover-buttons"
                    >
                      <button
                        class="gds-button-content"
                        color="secondary"
                        data-testid="cancel"
                        tabindex="0"
                        type="button"
                      >
                        <div
                          class="button-label-container"
                        >
                          Cancel
                        </div>
                      </button>
                      <button
                        class="gds-button-content"
                        color="primary"
                        data-testid="confirm"
                        tabindex="0"
                        type="button"
                      >
                        <div
                          class="button-label-container"
                        >
                          Confirm
                        </div>
                      </button>
                    </div>
                  </div>
                  has ref
                </div>
              </div>
            </div>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;

exports[`LineEditableDataTable component LineEditableDataTable handles delete with previous defined editedrowindex 1`] = `
<div>
  <div>
    <table
      class="gds-table"
    >
      <thead>
        <tr>
          <th>
            <div
              class="table--header-value"
              tabindex="-1"
            >
              <span
                class="table--header-value--title right"
              >
                Number
              </span>
            </div>
          </th>
          <th>
            <div
              class="table--header-value"
              tabindex="-1"
            >
              <span
                class="table--header-value--title center"
              >
                Actions
              </span>
            </div>
          </th>
        </tr>
      </thead>
      <tbody>
        <tr
          class=""
          tabindex="-1"
        >
          <td
            class=""
          >
            <div
              class="gds-field-group"
            >
              <span
                class="gds-amount-field amount-field-read-only table--cell-value--number-readonly"
                id="number"
                name="number"
              >
                1
              </span>
            </div>
          </td>
          <td
            class=""
          >
            <div
              class="table--cell--value--button"
            >
              <button
                class="gds-button-content"
                color="reversed"
                tabindex="0"
                type="button"
              >
                <div
                  class="button-icon-container"
                >
                  <i
                    class="button-icon icon"
                    icon="fal,trash-alt"
                  />
                </div>
              </button>
              <div>
                Delete
              </div>
              <div>
                <div>
                  ControlleMenu
                  closed
                  <div
                    class="gds-popover-container"
                  >
                    <div
                      class="popover-title"
                    >
                      Delete?
                    </div>
                    <div
                      class="popover-buttons"
                    >
                      <button
                        class="gds-button-content"
                        color="secondary"
                        data-testid="cancel"
                        tabindex="0"
                        type="button"
                      >
                        <div
                          class="button-label-container"
                        >
                          Cancel
                        </div>
                      </button>
                      <button
                        class="gds-button-content"
                        color="primary"
                        data-testid="confirm"
                        tabindex="0"
                        type="button"
                      >
                        <div
                          class="button-label-container"
                        >
                          Confirm
                        </div>
                      </button>
                    </div>
                  </div>
                  has ref
                </div>
              </div>
            </div>
          </td>
        </tr>
        <tr
          class=""
          tabindex="-1"
        >
          <td
            class=""
          >
            <div
              class="gds-field-group"
            >
              <input
                class="gds-amount-field"
                inputmode="numeric"
                name="number"
                placeholder="NUMBER"
                type="text"
                value="2"
              />
            </div>
          </td>
          <td
            class=""
          >
            <div
              class="table--cell--value--button"
            />
          </td>
        </tr>
      </tbody>
    </table>
    <div>
      <div
        class="gds-modal-overlay"
      >
        <div
          class="modal-content size-lg"
        >
          <div
            class="modal-header"
          >
            <div
              class="modal-title"
            >
              Edit row
            </div>
          </div>
          <div
            class="modal-body"
          >
            <div
              class="gds-layout-container"
            >
              <div
                class="gds-layout-col"
              >
                <div
                  class="gds-field-group"
                >
                  <label
                    class="field-label"
                  >
                    Number
                    <span
                      class="field-label-mandatory"
                    >
                      *
                    </span>
                  </label>
                  <input
                    class="gds-amount-field"
                    inputmode="numeric"
                    name="number"
                    placeholder="NUMBER"
                    type="text"
                    value="2"
                  />
                </div>
              </div>
            </div>
          </div>
          <div
            class="modal-footer"
          >
            <button
              class="gds-button-content"
              color="secondary"
              tabindex="0"
              type="button"
            >
              <div
                class="button-label-container"
              >
                Cancel
              </div>
            </button>
            <button
              class="gds-button-content"
              color="primary"
              tabindex="0"
              type="button"
            >
              <div
                class="button-label-container"
              >
                Submit
              </div>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`LineEditableDataTable component LineEditableDataTable handles download 1`] = `
<div>
  <div>
    <table
      class="gds-table"
    >
      <thead>
        <tr>
          <th>
            <div
              class="table--header-value"
              tabindex="-1"
            >
              <span
                class="table--header-value--title right"
              >
                Number
              </span>
            </div>
          </th>
          <th>
            <div
              class="table--header-value"
              tabindex="-1"
            >
              <span
                class="table--header-value--title center"
              >
                Actions
              </span>
            </div>
          </th>
        </tr>
      </thead>
      <tbody>
        <tr
          class=""
          tabindex="-1"
        >
          <td
            class=""
          >
            <div
              class="gds-field-group"
            >
              <span
                class="gds-amount-field amount-field-read-only table--cell-value--number-readonly"
                id="number"
                name="number"
              >
                1
              </span>
            </div>
          </td>
          <td
            class=""
          >
            <div
              class="table--cell--value--button"
            >
              <button
                class="gds-button-content"
                color="reversed"
                tabindex="0"
                type="button"
              >
                <div
                  class="button-icon-container"
                >
                  <i
                    class="button-icon icon"
                    icon="fal,arrow-down-to-line"
                  />
                </div>
              </button>
              <div>
                Download
              </div>
            </div>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;

exports[`LineEditableDataTable component LineEditableDataTable handles edit and submit 1`] = `
<div>
  <div>
    <table
      class="gds-table"
    >
      <thead>
        <tr>
          <th>
            <div
              class="table--header-value"
              tabindex="-1"
            >
              <span
                class="table--header-value--title right"
              >
                Number
              </span>
            </div>
          </th>
          <th>
            <div
              class="table--header-value"
              tabindex="-1"
            >
              <span
                class="table--header-value--title center"
              >
                Actions
              </span>
            </div>
          </th>
        </tr>
      </thead>
      <tbody>
        <tr
          class=""
          tabindex="-1"
        >
          <td
            class=""
          >
            <div
              class="gds-field-group"
            >
              <span
                class="gds-amount-field amount-field-read-only table--cell-value--number-readonly"
                id="number"
                name="number"
              >
                1
              </span>
            </div>
          </td>
          <td
            class=""
          >
            <div
              class="table--cell--value--button"
            >
              <button
                class="gds-button-content"
                color="reversed"
                tabindex="0"
                type="button"
              >
                <div
                  class="button-icon-container"
                >
                  <i
                    class="button-icon icon"
                    icon="fal,edit"
                  />
                </div>
              </button>
              <div>
                Edit
              </div>
            </div>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;

exports[`LineEditableDataTable component LineEditableDataTable handles non deletable 1`] = `
<div>
  <div>
    <table
      class="gds-table"
    >
      <thead>
        <tr>
          <th>
            <div
              class="table--header-value"
              tabindex="-1"
            >
              <span
                class="table--header-value--title right"
              >
                Number
              </span>
            </div>
          </th>
          <th>
            <div
              class="table--header-value"
              tabindex="-1"
            >
              <span
                class="table--header-value--title center"
              >
                Actions
              </span>
            </div>
          </th>
        </tr>
      </thead>
      <tbody>
        <tr
          class=""
          tabindex="-1"
        >
          <td
            class=""
          >
            <div
              class="gds-field-group"
            >
              <span
                class="gds-amount-field amount-field-read-only table--cell-value--number-readonly"
                id="number"
                name="number"
              >
                1
              </span>
            </div>
          </td>
          <td
            class=""
          >
            <div
              class="table--cell--value--button"
            />
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;

exports[`LineEditableDataTable component LineEditableDataTable handles non downloadable 1`] = `
<div>
  <div>
    <table
      class="gds-table"
    >
      <thead>
        <tr>
          <th>
            <div
              class="table--header-value"
              tabindex="-1"
            >
              <span
                class="table--header-value--title right"
              >
                Number
              </span>
            </div>
          </th>
          <th>
            <div
              class="table--header-value"
              tabindex="-1"
            >
              <span
                class="table--header-value--title center"
              >
                Actions
              </span>
            </div>
          </th>
        </tr>
      </thead>
      <tbody>
        <tr
          class=""
          tabindex="-1"
        >
          <td
            class=""
          >
            <div
              class="gds-field-group"
            >
              <span
                class="gds-amount-field amount-field-read-only table--cell-value--number-readonly"
                id="number"
                name="number"
              >
                1
              </span>
            </div>
          </td>
          <td
            class=""
          >
            <div
              class="table--cell--value--button"
            />
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;

exports[`LineEditableDataTable component LineEditableDataTable handles non editable content 1`] = `
<div>
  <div>
    <table
      class="gds-table"
    >
      <thead>
        <tr>
          <th>
            <div
              class="table--header-value"
              tabindex="-1"
            >
              <span
                class="table--header-value--title right"
              >
                Number
              </span>
            </div>
          </th>
          <th>
            <div
              class="table--header-value"
              tabindex="-1"
            >
              <span
                class="table--header-value--title center"
              >
                actionColumn
              </span>
            </div>
          </th>
        </tr>
      </thead>
      <tbody>
        <tr
          class=""
          tabindex="-1"
        >
          <td
            class=""
          >
            <div
              class="gds-field-group"
            >
              <span
                class="gds-amount-field amount-field-read-only table--cell-value--number-readonly"
                id="number"
                name="number"
              >
                1
              </span>
            </div>
          </td>
          <td
            class=""
          >
            <div
              class="table--cell--value--button"
            />
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;

exports[`LineEditableDataTable component LineEditableDataTable handles sort 1`] = `
<div>
  <div>
    <table
      class="gds-table"
    >
      <thead>
        <tr>
          <th>
            <div
              class="table--header-value table--header-value--sortable"
              tabindex="0"
            >
              <span
                class="table--header-value--title right"
              >
                Number
              </span>
              <span
                class="table--header-value--sorter"
              >
                <i
                  class="icon"
                  icon="fas,sort"
                />
              </span>
            </div>
            <div>
              Click to sort
            </div>
          </th>
        </tr>
      </thead>
      <tbody>
        <tr
          class="no-data"
        >
          <td
            colspan="1"
          >
            Loading
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;

exports[`LineEditableDataTable component LineEditableDataTable renders 1`] = `
<div>
  <div>
    <table
      class="gds-table"
    >
      <thead>
        <tr>
          <th>
            <div
              class="table--header-value"
              tabindex="-1"
            >
              <span
                class="table--header-value--title right"
              >
                Number
              </span>
            </div>
          </th>
        </tr>
      </thead>
      <tbody>
        <tr
          class=""
          tabindex="-1"
        >
          <td
            class=""
          >
            <div
              class="gds-field-group"
            >
              <span
                class="gds-amount-field amount-field-read-only table--cell-value--number-readonly"
                id="number"
                name="number"
              >
                1
              </span>
            </div>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;

exports[`LineEditableDataTable component LineEditableDataTable renders while loading 1`] = `
<div>
  <div>
    <table
      class="gds-table"
    >
      <thead>
        <tr>
          <th>
            <div
              class="table--header-value"
              tabindex="-1"
            >
              <span
                class="table--header-value--title right"
              >
                Number
              </span>
            </div>
          </th>
        </tr>
      </thead>
      <tbody>
        <tr
          class="no-data"
        >
          <td
            colspan="1"
          >
            Loading
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;

exports[`LineEditableDataTable component LineEditableDataTable renders with no data 1`] = `
<div>
  <div>
    <table
      class="gds-table"
    >
      <thead>
        <tr>
          <th>
            <div
              class="table--header-value"
              tabindex="-1"
            >
              <span
                class="table--header-value--title right"
              >
                Number
              </span>
            </div>
          </th>
        </tr>
      </thead>
      <tbody>
        <tr
          class="no-data"
        >
          <td
            colspan="1"
          >
            <div
              class="no-data-container"
            >
              <div
                class="no-data-text"
              >
                No data
              </div>
              <i
                class="no-data-icon icon"
                icon="fal,inbox"
              />
            </div>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;

exports[`LineEditableDataTable component LineEditableDataTable with custom modal button 1`] = `
<div>
  <div>
    <table
      class="gds-table"
    >
      <thead>
        <tr>
          <th>
            <div
              class="table--header-value"
              tabindex="-1"
            >
              <span
                class="table--header-value--title right"
              >
                Number
              </span>
            </div>
          </th>
        </tr>
      </thead>
      <tbody>
        <tr
          class="no-data"
        >
          <td
            colspan="1"
          >
            <div
              class="no-data-container"
            >
              <div
                class="no-data-text"
              >
                No data
              </div>
              <i
                class="no-data-icon icon"
                icon="fal,inbox"
              />
            </div>
          </td>
        </tr>
      </tbody>
    </table>
    <button
      class="gds-button-content gds-table-new-line"
      color="primary"
      tabindex="0"
      type="button"
    >
      <div
        class="button-label-container"
      >
        Add row
      </div>
    </button>
  </div>
</div>
`;
